js/compressed-token/src/actions/compress.ts

```diff
@@ -0,0 +1,71 @@
+import {
+    ConfirmOptions,
+    PublicKey,
+    Signer,
+    TransactionSignature,
+} from '@solana/web3.js';
+import {
+    bn,
+    defaultTestStateTreeAccounts,
+    sendAndConfirmTx,
+    buildAndSignTx,
+    Rpc,
+} from '@lightprotocol/stateless.js';
+
+import { BN } from '@coral-xyz/anchor';
+
+import { dedupeSigner, getSigners } from './common';
+import { CompressedTokenProgram } from '../program';
+
+/**
+ * Compress SPL tokens
+ *
+ * @param rpc                   Rpc connection to use
+ * @param payer                 Payer of the transaction fees
+ * @param mint                  Mint of the compressed token
+ * @param amount                Number of tokens to transfer
+ * @param owner                 Owner of the compressed tokens. Must be Signer
+ *                              unless `multiSigners` is provided.
+ * @param sourceTokenAccount    Source (associated) token account
+ * @param toAddress             Destination address of the recipient
+ * @param merkleTree            State tree account that the compressed tokens
+ *                              should be inserted into. Defaults to a default
+ *                              state tree account.
+ * @param multiSigners          Signing accounts if `currentOwner` is a multisig
+ * @param confirmOptions        Options for confirming the transaction
+ *
+ *
+ * @return Signature of the confirmed transaction
+ */
+export async function compress(
+    rpc: Rpc,
+    payer: Signer,
+    mint: PublicKey,
+    amount: number | BN,
+    owner: Signer | PublicKey,
+    sourceTokenAccount: PublicKey,
+    toAddress: PublicKey,
+    merkleTree: PublicKey = defaultTestStateTreeAccounts().merkleTree,
+    multiSigners: Signer[] = [],
+    confirmOptions?: ConfirmOptions,
+): Promise<TransactionSignature> {
+    const [currentOwnerPublicKey, signers] = getSigners(owner, multiSigners);
```

Comment:
I am a bit confused by multiple signers and separate payer and owner, but need to think about how I would do it differently.

Reply:
yea multiSigners was a leftover from dealing with multisigs, (i.e. `mintAuthority` and `owner` being a multisig)
we'll have to deal with that in a future pr - straightforward support for multisigs.